cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringStore.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.tethering;
+
+import com.google.common.collect.ImmutableList;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.inject.Inject;
+import io.cdap.cdap.api.dataset.lib.CloseableIterator;
+import io.cdap.cdap.spi.data.StructuredRow;
+import io.cdap.cdap.spi.data.StructuredTable;
+import io.cdap.cdap.spi.data.table.field.Field;
+import io.cdap.cdap.spi.data.table.field.Fields;
+import io.cdap.cdap.spi.data.table.field.Range;
+import io.cdap.cdap.spi.data.transaction.TransactionRunner;
+import io.cdap.cdap.spi.data.transaction.TransactionRunners;
+import io.cdap.cdap.store.StoreDefinition;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Store for tethering data.
+ */
+public class TetheringStore {
+  private static final Gson GSON = new GsonBuilder().create();
+
+  private final TransactionRunner transactionRunner;
+
+  @Inject
```

Comment:
do we need inject if we are just newing?

Reply:
not necessary, removed

Reply:
It's injected in a couple of other places. I'll inject the dependency everywhere to be consistent.
```
2021-12-07T02:01:29.3105193Z 3) Could not find a suitable constructor in io.cdap.cdap.internal.tethering.TetheringStore. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
2021-12-07T02:01:29.3107609Z   at io.cdap.cdap.internal.tethering.TetheringStore.class(TetheringStore.java:43)
2021-12-07T02:01:29.3109727Z   while locating io.cdap.cdap.internal.tethering.TetheringStore
2021-12-07T02:01:29.3111607Z     for parameter 1 at io.cdap.cdap.internal.tethering.TetheringClientHandler.<init>(TetheringClientHandler.java:51)
2021-12-07T02:01:29.3117164Z   at io.cdap.cdap.app.guice.AppFabricServiceRuntimeModule$AppFabricServiceModule.configure(AppFabricServiceRuntimeModule.java:399) (via modules: com.google.inject.util.Modules$OverrideModule -> io.cdap.cdap.internal.guice.AppFabricTestModule -> com.google.inject.util.Modules$CombinedModule -> io.cdap.cdap.app.guice.AppFabricServiceRuntimeModule$AppFabricServiceModule)
2021-12-07T02:01:29.3120558Z 
2021-12-07T02:01:29.3120861Z 3 errors
2021-12-07T02:01:29.3122240Z 	at com.google.inject.internal.Errors.throwCreationExceptionIfErrorsExist(Errors.java:466)
```